File: app/src/test/java/com/egecius/rxjava2_demo_2/rx/assertions/AssertionsExamples.java
package com.egecius.rxjava2_demo_2.rx.assertions;

import io.reactivex.Observable;

public class AssertionsExamples {

    private static final String EXCEPTION_MESSAGE = "egis";
    private static final EgisException EGIS_EXCEPTION = new EgisException(EXCEPTION_MESSAGE);

    public EgisException getException() {
        return EGIS_EXCEPTION;
    }

    public String getExceptionMessage() {
        return EXCEPTION_MESSAGE;
    }

    Observable<Integer> getJust() {
        return Observable.just(1);
    }

    Observable<Integer> emitThreeIntegersAndFail() {
        return Observable.create(emitter -> {
            for (int i = 0; i < 3; i++) {
                emitter.onNext(i);
            }

            emitter.onError(EGIS_EXCEPTION);
        });
    }

    Observable<Integer> emitWithoutCompletion(int item1, int item2) {
        return Observable.create(e -> {
            e.onNext(item1);
            e.onNext(item2);
        });
    }
}

File: app/src/test/java/com/egecius/rxjava2_demo_2/rx/assertions/AssertionsExamples.kt
package com.egecius.rxjava2_demo_2.rx.assertions

import com.egecius.rxjava2_demo_2.rx.EgisException

import io.reactivex.Observable

class AssertionsExamples {

    val exception: EgisException
        get() = EGIS_EXCEPTION

    val exceptionMessage: String
        get() = EXCEPTION_MESSAGE

    val just: Observable<Int>
        get() = Observable.just(1)

    fun emitThreeIntegersAndFail(): Observable<Int> {
        return Observable.create { emitter ->
            for (i in 0..2) {
                emitter.onNext(i)
            }

            emitter.onError(EGIS_EXCEPTION)
        }
    }

    fun emitWithoutCompletion(item1: Int, item2: Int): Observable<Int> {
        return Observable.create { e ->
            e.onNext(item1)
            e.onNext(item2)
        }
    }

    companion object {

        private val EXCEPTION_MESSAGE = "egis"
        private val EGIS_EXCEPTION = EgisException(EXCEPTION_MESSAGE)
    }
}
